vendor/brave-ios/Ads/BATBraveAds.mm

```diff
@@ -386,7 +386,7 @@ - (void)reportLoadedPageWithURL:(NSURL *)url innerText:(NSString *)text
 {
   if (![self isAdsServiceRunning]) { return; }
   const auto urlString = base::SysNSStringToUTF8(url.absoluteString);
-  ads->OnPageLoaded(urlString, base::SysNSStringToUTF8(text));
+  ads->OnPageLoaded(urlString, urlString, base::SysNSStringToUTF8(text));
```

Comment:
@kylehickinson For iOS we need to pass the original url, i.e. before any redirects as the first parameter. We can then need to add a call to `ads_->OnPushState(urlString)` when page history is updated via pushState

Reply:
Not sure how this will work with WebKit, will need additional research

Reply:
I will raise a separate ticket for iOS as no need to block desktop